packages/mds-trip-processor/src/proc-trip.ts

```diff
@@ -0,0 +1,211 @@
+import db from '@mds-core/mds-db'
+import cache from '@mds-core/mds-cache'
+import log from '@mds-core/mds-logger'
+import { calcDistance } from '@mds-core/mds-utils'
+
+import { TripEvent, TripEntry, TripTelemetry } from '@mds-core/mds-types'
+import config from './config'
+import { dataHandler } from './proc'
+
+/*
+    Trip processor that runs inside a Kubernetes pod, activated via cron job.
+    Aggregates event/telemety data into binned trips at a set interval. As trips
+    are processed caches are cleaned.
+
+    Processed trips are added to a postgres table:
+
+        REPORTS_DEVICE_TRIPS:
+          PRIMARY KEY = (provider_id, device_id, trip_id)
+          VALUES = trip_data
+*/
+
+async function processTrip(
+  provider_id: string,
+  device_id: string,
+  trip_id: string,
+  tripEvents: TripEvent[],
+  curTime: number
+): Promise<boolean> {
+  /*
+    Add telemetry and meta data into database when a trip ends
+
+    Examples:
+
+        1) trip duration
+        2) trip length
+
+    We must compute these metrics here due to the potential of up to 24hr delay of telemetry data
+  */
+
+  // Validation steps
+  // TODO: make checks more robust
+  if (tripEvents.length < 2) {
+    log.info('No trip end seen yet')
+    return false
+  }
+
+  // Process anything where the last event timestamp is more than 24 hours old
+  tripEvents.sort((a: { timestamp: number }, b: { timestamp: number }) => {
```

Comment:
Nit: This could probably be done with a `Pick`...also I believe we have a `Timestamp` type alias